examples/Main.hs

```diff
@@ -250,6 +250,16 @@ plutarchTests =
             let v2 = [("IOHK", [1, 2, 3]), ("Plutus", [9, 8, 7])]
             plift' (pconstant @(PBuiltinList (PBuiltinPair PString (PBuiltinList PInteger))) v2) @?= Right v2
         ]
+    , testCase "True && False ≡ False" $ equal (pcon PTrue #&& pcon PFalse) (pcon PFalse)
+    , testCase "False && True ≡ False" $ equal (pcon PFalse #&& pcon PTrue) (pcon PFalse)
+    , testCase "False && perror ≡ False" $ equal (pcon PFalse #&& perror) (pcon PFalse)
+    , testCase "fails: pand False perror" $ fails $ pand # pcon PFalse # perror
```

Comment:
This doesn't seem right. Per the semantics of `fails` it expects _either_ user error (`EvaluationError`) _or_ engine error/bug (`EvaluationException`). Even `EvaluationError` is not precise:

```haskell
data EvaluationError user internal
    = InternalEvaluationError internal
      -- ^ Indicates bugs.
    | UserEvaluationError user
      -- ^ Indicates user errors.
```

What we want is to test that the result is exactly a `perror` (and not general evaluation error/exception).

Reply:
This feels like something for another PR. I don't touch the implementation of `fails` here.

Reply:
Filed https://github.com/Plutonomicon/plutarch/issues/85